src/services/completions.ts

```diff
@@ -1957,7 +1957,7 @@ namespace ts.Completions {
     }
 
     function isFunctionLikeBodyKeyword(kind: SyntaxKind) {
-        return kind === SyntaxKind.AsyncKeyword || !isContextualKeyword(kind) && !isClassMemberCompletionKeyword(kind);
+        return kind === SyntaxKind.AsyncKeyword || kind === SyntaxKind.AwaitKeyword || !isContextualKeyword(kind) && !isClassMemberCompletionKeyword(kind);

```

Comment:
I don't think this is right fix. You want to create `AsyncFunctionLikeBodyKeywords` and add await in there to ensure that await is offered only in `async` functions? Also remove it from global completions?

Reply:
Also please add a test case.

Reply:
I dont think i have enough skill to remove await from global keywords and create AsyncFunctionLikeBodyKeywords.
I had no idea about how to adding AsyncFunctionLikeBodyKeywords

Reply:
BTW, For now that maybe a shortcut to adding `await` keyword to inside of function. I am working so smooth with that fix by local build.

Reply:
I think that...

- Typescript 3.2.2 (Quick Fix)

  1. Add `await` inside of function for quick fix
  2. At least people can use it as quickly as possible

Then

- Typescript 3.3.0 (Deep Fix)

  1. Create AsyncFunctionLikeBodyKeywords
  2. Remove `await` from `globalKeywords`
  3. Remove `await` from `normal function`
  4. Add `await` inside of `async function` only

Reply:
@DanielRosenwasser To triage on if  we want this fix as suggested by @Cow258 for 3.2.2?

Reply:
> Also please add a test case.

After this fix `await` keyword will be offered global and inside of function.
![await](https://user-images.githubusercontent.com/14261588/49464834-02e12780-f837-11e8-9c6f-0eeac2759749.gif)

Reply:
It sounds like you already have the right direction for the fix, so let's just do that today and get it into 3.2.2.